notificationdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/notification/MainActivity.java

```diff
@@ -0,0 +1,218 @@
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.notification;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+
+import org.kaaproject.kaa.client.notification.NotificationListener;
+import org.kaaproject.kaa.client.notification.NotificationTopicListListener;
+import org.kaaproject.kaa.common.endpoint.gen.Topic;
+import org.kaaproject.kaa.demo.notification.entity.TopicPojo;
+import org.kaaproject.kaa.demo.notification.fragment.NotificationDialogFragment;
+import org.kaaproject.kaa.demo.notification.fragment.NotificationFragment;
+import org.kaaproject.kaa.demo.notification.fragment.OnFragmentUpdateEvent;
+import org.kaaproject.kaa.demo.notification.fragment.TopicFragment;
+import org.kaaproject.kaa.demo.notification.kaa.KaaManager;
+import org.kaaproject.kaa.demo.notification.storage.TopicStorage;
+import org.kaaproject.kaa.demo.notification.util.NotificationConstants;
+import org.kaaproject.kaa.demo.notification.util.TopicHelper;
+import org.kaaproject.kaa.schema.sample.notification.SecurityAlert;
+
+import java.util.List;
+
+import static org.kaaproject.kaa.demo.notification.util.TopicHelper.addNotificationToTopic;
+
+/**
+ * The implementation of the {@link FragmentActivity} class.
+ * Use multithreading for starting Kaa client
+ * Init Notification and Topic listener, which can get information from server
+ * Manage all views.
+ * <p>
+ * Tip: you can use Service with this AsyncTask and send all Kaa Notification in mobile notification area
+ */
+public class MainActivity extends AppCompatActivity implements TopicFragment.OnTopicClickedListener {
+
+    private KaaManager manager;
+    private NotificationListener notificationListener;
+    private NotificationTopicListListener topicListener;
+
+    @Override
+    public void onCreate(Bundle savedInstance) {
+        super.onCreate(savedInstance);
+        setContentView(R.layout.activity_main);
+
+        setUpToolbar();
+        showProgress();
+        startKaa();
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
+    }
+
+    public void startKaa() {
+        kaaTask.execute();
+    }
+
+    // Tip: you can use it from Service
+    private AsyncTask<Void, Void, Void> kaaTask = new AsyncTask<Void, Void, Void>() {
```

Comment:
@maxml AsyncTask is used here because staring Kaa client takes some time and it's better to not block the UI thread and do it asynchronously in this case